src/directive.js

```diff
@@ -27,11 +27,16 @@
             link: function (scope, elm, attrs, ctrl) {
                 scope.widgetId = null;
 
-                if(ctrl && ng.isDefined(attrs.required)){
+                if (ctrl && ng.isDefined(attrs.required)) {
                     scope.$watch('required', validate);
                 }
 
                 var removeCreationListener = scope.$watch('key', function (key) {
+                    // Abort if undefined or null
+                    if (ng.isUndefined(key) || key === null) {
+                        return;
```

Comment:
This early return prevents the service from throwing an error when the key is undefined. Check the [failing test](https://github.com/VividCortex/angular-recaptcha/blob/a187773e5e730cc54d43379be13e79e9d83f7a50/tests/directive_test.js#L35-L38).